models/classes/user/Lti1p3User.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2021 (original work) Open Assessment Technologies SA;
+ */
+
+declare(strict_types=1);
+
+namespace oat\taoLti\models\classes\user;
+
+use oat\generis\model\GenerisRdf;
+use oat\generis\model\OntologyRdfs;
+use oat\taoLti\models\classes\LtiInvalidVariableException;
+use oat\taoLti\models\classes\LtiLaunchData;
+use oat\taoLti\models\classes\LtiRoles;
+use oat\taoLti\models\classes\LtiUtils;
+use Zend\ServiceManager\ServiceLocatorAwareInterface;
+use Zend\ServiceManager\ServiceLocatorAwareTrait;
+use oat\oatbox\user\UserLanguageService;
+
+/**
+ * Authentication adapter interface to be implemented by authentication methodes
+ *
+ * @access public
+ * @author Joel Bout, <joel@taotesting.com>
+ * @package taoLti
+ */
+class Lti1p3User extends \common_user_User implements ServiceLocatorAwareInterface, \JsonSerializable, LtiUserInterface
```

Comment:
Any chance to reuse `LtiUser`'s methods here by either
1. Extracting the common ones into dedicated services (e.g. a serializer)
2. Including the LtiUser object as a dependency of this new class, delegating some method calls to the internal object, having sort of a composition of both.

Reply:
done